### PR TITLE
Fix firefox file upload

### PR DESCRIPTION
**Pull request by vstdolittle**
## Summary

Fix firefox file upload not working. Refactor code slightly to improve code quality

### Changed

- Moved part of the form logic to its own component. Cleans up the code slightly.

### Fixed

- File upload not working for Firefox. Firefox uses default browser behaviour where the button click inside a form triggers the onsubmit event

### CODE REVIEW

File: Source/SelfService/Web/microservice/base/components/htmlHiddenForm.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * @returns
+ */
+export default function HTMLHiddenForm(props: HTMLHiddenFormProps) {
+
+    /**
+     *
+     * @param event Serves submit event and formdata to client
+     */
+    const onFileAdd = (event) => {
+        event.preventDefault();
```
[the remark this answers is not in the document]
Comment (vstdolittle) [reply Source/SelfService/Web/microservice/base/components/htmlHiddenForm.tsx:29]: it was actually the cancelable: true flag combined with this statement that fixed the issue. cancelable false indicated preventDefault would be ignored.

File: Source/SelfService/Web/microservice/base/configuration.tsx [hunk 2]
```diff
@@ -188,11 +174,7 @@ export const Configuration: React.FunctionComponent<ConfigurationProps> = (props
                             Download config files yaml
                         </DownloadButton>
                     </Grid>
-                    <form method="put" id="config-file-selector-form" hidden onSubmit={onFileAdd}>
-                        <input type="file" id="file-selector" name='file' onChange={onFileSelect} />
-                        <input type="submit" id="file-submit" value="Submit" />
-                    </form>
-
+                    <HTMLHiddenForm onFileAdd={onFileAdd} onFileSelect={onFileSelect}/>
```
[the remark this answers is not in the document]
Comment (pavsaund) [reply Source/SelfService/Web/microservice/base/configuration.tsx]: Done ✅

File: Source/SelfService/Web/microservice/base/configuration.tsx [hunk 3]
```diff
@@ -137,7 +129,7 @@ export const Configuration: React.FunctionComponent<ConfigurationProps> = (props
                     setConfigFileModalVisibility(false);
                 }}
                 onAdd={()=>{
-                    const event = new Event('submit', {bubbles: true,   cancelable: false});
+                    const event = new Event('submit', {bubbles: true,   cancelable: true});
```
[the remark this answers is not in the document]
Comment (pavsaund) [reply Source/SelfService/Web/microservice/base/configuration.tsx]: I learned about the `useImperativeForm` hook, which makes a lot of sense and combined with `useRef` and `forwardRef` removed the need to reference dom elements outside of react's rendering cycles.

- official docs: https://reactjs.org/docs/hooks-reference.html#useimperativehandle
- SO reference: https://stackoverflow.com/questions/53561913/react-forwarding-multiple-refs

**Comment by vstdolittle** [comment]
i like these changes and how you solved using the correct reference for emitting the event. useImperative is a good one. Also using the ref instead of a hook around the whole form was a good choice. thats where i got stuck the other day. i pushed two small changes to reduce the lines of code and unused block.

**Review by pavsaund — APPROVED** [review]
Awesomesauce! let's do this! :)

**Pull request merged**